demos/social_distance_demo/cpp/main.cpp

```diff
@@ -0,0 +1,896 @@
+        if (Sec{1} <= timeDuration || context.drawersContext.updateTime == context.t0) {
+            context.drawersContext.outThroughput.str("");
+            context.drawersContext.outThroughput << std::fixed << std::setprecision(1)
+                << static_cast<float>(context.drawersContext.framesAfterUpdate) / timeDuration.count() << "FPS";
+            context.drawersContext.framesAfterUpdate = 0;
+            context.drawersContext.updateTime = localT1;
+        }
+        cv::putText(mat, context.drawersContext.outThroughput.str(), cv::Point2f(15, 35), cv::FONT_HERSHEY_TRIPLEX, 0.7, cv::Scalar{255, 255, 255});
+
+        context.drawersContext.presenter.drawGraphs(mat);
+        if (!FLAGS_no_show) {
+            cv::imshow("Detection results", firstGridIt->second.getMat());
+            context.drawersContext.prevShow = std::chrono::steady_clock::now();
+            const int key = cv::waitKey(context.drawersContext.pause);
+            if (key == 27 || 'q' == key || 'Q' == key || !context.isVideo) {
+                try {
+                    std::shared_ptr<Worker>(context.drawersContext.drawersWorker)->stop();
+                }
+                catch (const std::bad_weak_ptr&) {}
+            }
+            else if (key == 32) {
+                context.drawersContext.pause = (context.drawersContext.pause + 1) & 1;
+            }
+            else {
+                context.drawersContext.presenter.handleKey(key);
+            }
+        }
+        else {
+            if (!context.isVideo) {
+                try {
+                    std::shared_ptr<Worker>(context.drawersContext.drawersWorker)->stop();
+                }
+                catch (const std::bad_weak_ptr&) {}
+            }
+        }
+        firstGridIt->second.clear();
+        gridMats.emplace((--gridMats.end())->first + 1, firstGridIt->second);
+        gridMats.erase(firstGridIt);
+    }
+    context.drawersContext.drawerMutex.unlock();
+}
+
+void ResAggregator::process() {
+    Context& context = static_cast<ReborningVideoFrame*>(sharedVideoFrame.get())->context;
+    context.trackersContext.trackerMutex.lock();
+
+    context.freeDetectionInfersCount += context.detectorsInfers.inferRequests.lockedSize();
+    context.frameCounter++;
+    auto frameID = sharedVideoFrame->frameId;
+    unsigned sourceID = sharedVideoFrame->sourceID;
+    auto& data = context.trackersContext.processedData[sourceID];
+    auto trackingDataIt = data.find(frameID);
+    if (data.end() == trackingDataIt) {
+        trackingDataIt = data.emplace(frameID, Context::trackingData{sharedVideoFrame, std::move(boxes), std::move(trackables)}).first;
+    }
+
+    auto& lastProcessedFrameId = context.trackersContext.lastProcessIDs[sourceID];
+    auto& personTracker = context.trackersContext.personTracker[sourceID];
+    auto firstTrackingData = data.begin();
+    if (lastProcessedFrameId == firstTrackingData->first) {
```

Comment:
If it is true than frame order is preserved and it is known `personTracker` has latest state to take a new frame. But if it's false, the object is discarded because it won't `tryPush()` a `Drawer`.

Reply:
But before this check, all needed info (frame, boxes, trackables) is stored int the `std::map`, and later it will be processed in it's turn, not discarded. Am I missing smth?

Reply:
If an instance of `ResAggregator` is skipped then the `map` will see more of data than there will be instances of `ResAggregator` (if you don't count skipped ones). There will be a tail of `trackingData` unprocessed in the end.

BTW
```C++
    auto trackingDataIt = data.find(frameID);
    if (data.end() == trackingDataIt) {
        trackingDataIt = data.emplace(frameID, Context::trackingData{sharedVideoFrame, std::move(boxes), std::move(trackables)}).first;
    }
```
is strange. `trackingDataIt` is not used if you find `frameID`. That means the data carried with current instance of `ResAggregator` is discarded.

Reply:
Yeah, I got it.
About 
```
 auto trackingDataIt = data.find(frameID);
    if (data.end() == trackingDataIt) {
        trackingDataIt = data.emplace(frameID, Context::trackingData{sharedVideoFrame, std::move(boxes), std::move(trackables)}).first;
    }
```
I think this check is redundant, we just can't have two equal frameID in different instances of ResAggregator.

Reply:
I think we can: there is exactly one `ResAggregator` per `ClassifiersAggregator`. There is one `ClassifiersAggregator` per `DetectionsProcessor`. But `DetectionsProcessor` may spawn itself
```C++
    // Run DetectionsProcessor for remaining persons
    if (!personRects.empty()) {
        tryPush(context.detectionsProcessorsContext.reidTasksWorker,
                std::make_shared<DetectionsProcessor>(sharedVideoFrame, std::move(classifiersAggregator), std::move(personRects)));
    }
```

Reply:
It reuses `classifiersAggregator`, so never mind

Reply:
So, if i proceesed all objects in std::map in cycle in order and `tryPush` each, these would solve the problem with unprocessed tail ? Smth like this:
```
void ResAggregator::process() {
    Context& context = static_cast<ReborningVideoFrame*>(sharedVideoFrame.get())->context;
    context.trackersContext.trackerMutex.lock();
    <some code>
    auto frameID = sharedVideoFrame->frameId;
    unsigned sourceID = sharedVideoFrame->sourceID;
    auto& data = context.trackersContext.processedData[sourceID];
    data.emplace(frameID, Context::trackingData{sharedVideoFrame, std::move(boxes), std::move(trackables)}).first;

    auto& lastProcessedFrameId = context.trackersContext.lastProcessIDs[sourceID];
    auto& personTracker = context.trackersContext.personTracker[sourceID];
    auto firstTrackingData = data.begin();
    if (lastProcessedFrameId == firstTrackingData->first) {
        for (auto& trackingData : data) {
            < tracking and drawing >
            tryPush(context.drawersContext.drawersWorker, std::make_shared<Drawer>(_sharedVideoFrame));
            ++lastProcessedFrameId;
        }
        data.clear();
    }
    context.trackersContext.trackerMutex.unlock();
}
```

Reply:
If `data` had 1, 3, 5 frames and `lastProcessedFrameId` is 1, this will track 3 without waiting for 2.

I think you can redefine `isReady()` returning false if the order is broken. That's what `Drawer` does but with time.

Reply:
Done